radare2: 2.9.0 -> 3.0.0, fix to use deps we provide

dtzWill opened the pull request:
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
radare2: 2.9.0 -> 3.0.0

dtzWill committed:
radare2: libuv and actually use deps we're providing

Particularly openssl and zlib.
There are others we may wish to provide but for now
let's at least use the ones in buildInputs :).

dtzWill committed:
r2-for-cutter: revert accidental change to version info

dtzWill commented:
Cutter updated in #48582 .

GrahamcOfBorg commented:
<!--REQUEST_ID=e2d0cf7a-c4e3-4eb9-a959-508eecae19ad-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=81f403ff-fe62-4c1a-a389-4a8ea48b3600)

The following builds were skipped because they don't evaluate on x86_64-darwin: radare2

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e2d0cf7a-c4e3-4eb9-a959-508eecae19ad-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=cf4d3ebf-4063-4746-bd15-959156f9337d)

Attempted: radare2

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/lib/libr_socket.so.3.0.0
shrinking /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/lib/libr_crypto.so.3.0.0
shrinking /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/lib/libr_hash.so.3.0.0
shrinking /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/lib/libr_util.so.3.0.0
gzipping man pages under /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/share/man/
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/lib  /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/bin
patching script interpreter paths in /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0
/nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0/bin/r2pm: interpreter directive changed from "/bin/sh" to "/nix/store/dsyc1z7ck08ga7l0b1jcxx35wj69qcii-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0...
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e2d0cf7a-c4e3-4eb9-a959-508eecae19ad-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=78f7c946-7728-45f9-a7cb-39494a3153cf)

Attempted: radare2

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/radiff2
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/radare2
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/rasm2
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/rahash2
gzipping man pages under /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/share/man/
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/lib  /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin
patching script interpreter paths in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0
/nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/r2pm: interpreter directive changed from "/bin/sh" to "/nix/store/r47p5pzx52m3n34vdgqpk5rvqgm0m24m-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0...
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=f5804654-1ce0-4676-b591-c6bc48ac8397-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=c8f5d7e4-f5b7-4b21-97af-7ad8e17d97eb)

The following builds were skipped because they don't evaluate on x86_64-darwin: r2-for-cutter, radare2

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=f5804654-1ce0-4676-b591-c6bc48ac8397-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=fa931822-e938-49fa-beb2-bfa8712d8109)

Attempted: r2-for-cutter, radare2

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/lib/libr_hash.so.2.9.0-git
shrinking /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/lib/libr_util.so.2.9.0-git
gzipping man pages under /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/share/man/
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/lib  /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/bin
patching script interpreter paths in /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14
/nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14/bin/r2pm: interpreter directive changed from "/bin/sh" to "/nix/store/dsyc1z7ck08ga7l0b1jcxx35wj69qcii-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14...
/nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14
/nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0
```
</p></details>

dtzWill committed:
radare2: fix update script to handle invocation from elsewhere

It looks like `with Path:` is supposed to be a context-manager
for changing directory but it doesn't seem to work and the only
way I've found to make that work is if we made our own and annotated it.

Instead just use nix search path to tell the expression where
to find the file it looks for.

Also re-use SCRIPT_DIR elsewhere instead of re-"computing" it.

dtzWill commented:
Small fix for the update script too while visiting this code :).

GrahamcOfBorg commented:
<!--REQUEST_ID=ac942589-3ca7-4bb1-9ac1-c78d6ddee7ef-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=a4fc731a-5b70-48b5-972e-f1708b512166)

The following builds were skipped because they don't evaluate on x86_64-darwin: r2-for-cutter, radare2

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=ac942589-3ca7-4bb1-9ac1-c78d6ddee7ef-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=36d14558-2972-4c29-a060-31fbf97e2a71)

Attempted: r2-for-cutter, radare2

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/lykcpvxrgpladb3d7z8ci2f8502cq5px-radare2-2018-08-14
/nix/store/087bm04fps8avkh88vzp5chyxxqy32fb-radare2-3.0.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=ac942589-3ca7-4bb1-9ac1-c78d6ddee7ef-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48581&attempt_id=5e78c5d2-596c-4420-9be6-fdd8e9ea514e)

Attempted: r2-for-cutter, radare2

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/r2agent
shrinking /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/rahash2
/nix/store/fck8hn8v64c19w7n5il1nvss9blx994k-radare2-2018-08-14
gzipping man pages under /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/share/man/
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
/nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0
stripping (with command strip and flags -S) in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/lib  /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin
patching script interpreter paths in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0
/nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0/bin/r2pm: interpreter directive changed from "/bin/sh" to "/nix/store/r47p5pzx52m3n34vdgqpk5rvqgm0m24m-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/0ir3wnwcdl62la5fvria8229xwcwgh1i-radare2-3.0.0...
```
</p></details>
